extensions/workouts/src/leaderboard.tsx

```diff
@@ -50,7 +50,8 @@ type AthleteActivities = {
 };
 
 function Leaderboard({ clubs }: { clubs: StravaSummaryClub[] }) {
-  const [clubId, setClubId] = useState(clubs[0].id.toString());
+  const { value, setValue: setClubId } = useLocalStorage("lastClubId", clubs[0].id.toString());
```

Comment:
Question (non-blocking): Did [useCachedState](https://developers.raycast.com/utilities/react-hooks/usecachedstate) not work for this use-case?